src/core/qgsvectorlayerutils.cpp

```diff
@@ -380,17 +380,24 @@ QgsFeature QgsVectorLayerUtils::createFeature( const QgsVectorLayer *layer, cons
     bool checkUnique = true;
 
     // in order of priority:
+    // 1. passed attribute value
```

Comment:
please add to the comment something like: "... if the field has not unique constraint"

Reply:
Sure! No problem!